rest_framework_json_api/renderers.py

```diff
@@ -273,6 +273,30 @@ def extract_relationships(cls, fields, resource, resource_instance):
         return utils.format_keys(data)
 
     @classmethod
+    def extract_relation_instance(cls, field_name, field, resource_instance, serializer):
```

Comment:
Thanks for thinking about it. The only alternative that I can think of is to have multiple returns. What do you think of something like the following:

```python
         try:
             return getattr(resource_instance, field_name)
         except AttributeError:
             pass

         try:
             # For ManyRelatedFields if `related_name` is not set
             # we need to access `foo_set` from `source`
             return getattr(resource_instance, field.child_relation.source)
         except AttributeError:
             pass

        <and so on...>
```

Is that any "better?" Or is it worse? It would remove the deep nesting even if the logic is functionally the same.